File: examples/documents/list-collection-documents.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Documents the user can access via a collection they're a member of.
export async function listCollectionDocuments(client: JsBaoClient, collectionId: string) {
  // #region example
  const { items, cursor } = await client.collections.listDocuments(collectionId, {
    limit: 50,
  });
  // #endregion example
  return { items, cursor };
}

File: examples/documents/list-collection-documents.swift
import JsBaoClient

// Documents the user can access via a collection they're a member of.
func listCollectionDocuments(client: JsBaoClient, collectionId: String) async throws {
  // #region example
  let page = try await client.collections.listDocuments(
    collectionId: collectionId,
    options: PaginationOptions(limit: 50)
  )
  let items = page.items
  // #endregion example
  _ = items
}
